src/webgl/loading.js

```diff
@@ -419,9 +436,13 @@ function parseObj(model, lines, materials= {}) {
           ) {
             model.faces.push(face);
             //same material for all vertices in a particular face
-            if (currentMaterial && materials[currentMaterial]) {
+            if (currentMaterial
+              && materials[currentMaterial]
+              && materials[currentMaterial].diffuseColor) {
               const materialDiffuseColor =
               materials[currentMaterial].diffuseColor;
+              //flag to track color or no color model
+              hasColoredVertices = true;
```

Comment:
I wonder if we can simplify this a bit to not need the set. If you add a new variable called `hasColorlessVertices` and set it to true in an `else` branch, could we check that only one or the other is true but not both? i.e. `hasColoredVertices !== hasColorlessVertices`. I think we just want to make sure that we either always add colors, or never do, so this would be making sure we always go through the same branch of the if statement here.

Reply:
@davepagurek Yes, that would serve the purpose too. But this would not allow any duplicating vertex also not to have color, meaning if we have diffuse color even once , every vertex should have a diffuse color but what if vertex 1 is assigned color once but is repeated in another face, and should we allow it not having color later on further occurance? How I address it here in a set is all unique vertices of a model if they have color even once , allow them not to have assigned a color later. Because the model would still be fully colored. let me know which way are we trying to achieve a colored model?